monai/networks/nets/segresnet_ds.py

```diff
@@ -425,3 +425,178 @@ def _forward(self, x: torch.Tensor) -> Union[None, torch.Tensor, list[torch.Tens
+            blocks = [
+                SegResBlock(
+                    spatial_dims=spatial_dims,
+                    in_channels=filters,
+                    kernel_size=kernel_size,
+                    norm=norm,
+                    act=act,
+                )
+                for _ in range(self.blocks_up[i])
+            ]
+            level_auto["blocks"] = nn.Sequential(*blocks)
+            if len(self.blocks_up) - i <= dsdepth:  # deep supervision heads
+                level_auto["head"] = Conv[Conv.CONV, spatial_dims](
+                    in_channels=filters,
+                    out_channels=out_channels,
+                    kernel_size=1,
+                    bias=True,
+                )
+            else:
+                level_auto["head"] = nn.Identity()
+            self.up_layers_auto.append(level_auto)
+
+        if n_up == 0:  # in a corner case of flat structure (no downsampling), attache a single head
+            level_auto = nn.ModuleDict(
+                {
+                    "upsample": nn.Identity(),
+                    "blocks": nn.Identity(),
+                    "head": Conv[Conv.CONV, spatial_dims](
+                        in_channels=filters,
+                        out_channels=out_channels,
+                        kernel_size=1,
+                        bias=True,
+                    ),
+                }
+            )
+            self.up_layers_auto.append(level_auto)
+
+    def _forward(
+        self, x: torch.Tensor, with_point, with_label
+    ) -> Union[None, torch.Tensor, list[torch.Tensor]]:
+        if self.preprocess is not None:
+            x = self.preprocess(x)
+
+        if not self.is_valid_shape(x):
+            raise ValueError(
+                f"Input spatial dims {x.shape} must be divisible by {self.shape_factor()}"
+            )
+
+        x_down = self.encoder(x)
+
+        x_down.reverse()
+        x = x_down.pop(0)
+
+        if len(x_down) == 0:
+            x_down = [torch.zeros(1, device=x.device, dtype=x.dtype)]
+
+        outputs: list[torch.Tensor] = []
+        outputs_auto: list[torch.Tensor] = []
+        x_ = x.clone()
+        if with_point:
+            i = 0
+            for level in self.up_layers:
+                x = level["upsample"](x)
+                x = x + x_down[i]
+                x = level["blocks"](x)
+
+                if len(self.up_layers) - i <= self.dsdepth:
+                    outputs.append(level["head"](x))
+                i = i + 1
+
+            outputs.reverse()
+        x = x_
+        if with_label:
+            i = 0
+            for level in self.up_layers_auto:
+                x = level["upsample"](x)
+                x = x + x_down[i]
+                x = level["blocks"](x)
+
+                if len(self.up_layers) - i <= self.dsdepth:
+                    outputs_auto.append(level["head"](x))
+                i = i + 1
+
+            outputs_auto.reverse()
+
+        # in eval() mode, always return a single final output
```

Comment:
does this line correct? According to the following logic, in eval() mode, will still return a list of tensor if len(output) != 1

Reply:
seems self.traning is True or False does not impact the return result, right?

Reply:
I removed it's logic